client/src/App.tsx

```diff
@@ -4,13 +4,14 @@ import { BrowserRouter, Route, Redirect, Switch } from 'react-router-dom';
 import Login from './pages/Login/Login';
 import Signup from './pages/SignUp/SignUp';
 import Dashboard from './pages/Dashboard/Dashboard';
-import { AuthProvider } from './context/useAuthContext';
+import { AuthProvider, useAuth } from './context/useAuthContext';
 import { SocketProvider } from './context/useSocketContext';
 import { SnackBarProvider } from './context/useSnackbarContext';
-
+import ProtectedRoute from './components/ProtectedRoute/ProtectedRoute.js';
 import './App.css';
 
 function App(): JSX.Element {
+  const { loggedInUser } = useAuth();
```

Comment:
If you move this hook inside of ProtectedRoute then you don't need to pass it as a prop everytime